crates/cli-flags/src/lib.rs

```diff
@@ -143,11 +149,15 @@ pub struct CommonOptions {
     pub wasi_modules: Option<WasiModules>,
 
     /// Generate jitdump file (supported on --features=profiling build)
-    #[clap(long, conflicts_with = "vtune")]
+    #[clap(long, conflicts_with_all = &["vtune", "perfmap"])]
     pub jitdump: bool,
 
-    /// Generate vtune (supported on --features=vtune build)
-    #[clap(long, conflicts_with = "jitdump")]
+    /// Generate perf mapping file
+    #[clap(long, conflicts_with_all = &["vtune", "jitdump"])]
+    pub perfmap: bool,
+
+    /// Generate vtune runtime information (supported on --features=vtune build)
+    #[clap(long, conflicts_with_all = &["jitdump", "perfmap"])]
     pub vtune: bool,
```

Comment:
We may be getting somewhat close to the limit of "now it's probably better to have `--profile=jitdump`" as opposed to a separate flag-per-strategy, but that's fine to defer to a different PR.